### ISSUE TITLE
Building packages on OS X silently fails during ncp call

### ISSUE THREAD

**Issue by davej** [electron-userland/electron-packager#150]
Using electron-packager v5.1.0 on OS X 10.10.5 and Node v4.1.1

```
➜ electron-packager . MyApp --overwrite --platform=darwin --arch=x64 --version=0.33.6

Packaging app for platform darwin x64 using electron v0.33.6
```

No errors, it just exits about 10 seconds after displaying the message "Packaging app for platform darwin x64 using electron v0.33.6" and there are no files written to my CWD.

I realise that this is very sparse on info but I'm not sure what other details I can give. When I last ran this command about three weeks ago it worked fine.

**Comment by malept** [comment]
Can you bisect which released version of electron-packager is the last one that works for you?

**Comment by davej** [comment]
I believe I was using 5.1.0 when it worked before I tried moving back to an earlier version but it still didn't work. Could this be something related to something in my app code base (I see symlink issues are mentioned a lot, I have symlinked modules).

**Comment by malept** [comment]
Could you try specifying an `--out` directory?

Otherwise, I think we're going to need reproduction steps on a minimal Electron app.

**Comment by davej** [comment]
Still isn't working with `--out`.

I'm doing a bit of debugging, seems to be [failing here](https://github.com/maxogden/electron-packager/blob/d985624694b2132127ea88bae6f770790234824f/common.js#L82) the callback function (`cb`) is never fired.

**Comment by malept** [comment]
Hmm. Very odd. I noticed that Travis CI only tests under Node 0.12. Let me add some more Node versions, and see if that surfaces anything.

**Comment by davej** [comment]
Possibly related to https://github.com/AvianFlu/ncp/issues/98?

**Comment by malept** [comment]
Possibly, but that's a really vague bug report with no follow-up.

**Comment by davej** [comment]
Yup.

FYI, tried a couple of other node versions, I'm getting the same issue with Node-v0.12.2 and iojs-v2.0.1.

**Comment by davej** [comment]
The actual copying of files works fine by the way, all the files are copied correctly, it's just that the callback isn't fired.

I put the callback in a `setTimeout` just to get it working and the app packaged itself correctly and runs as expected.

```javascript
  var ncpCallbackFired = false
  ncp(opts.dir, appPath, {filter: userIgnoreFilter(opts), dereference: true}, function(err) {
    ncpCallbackFired = true;
    cb(err);
  });
  // callback above isn't being fired by ncp for some reason,
  // so call it synthetically after 20 secs if it hasn't been fired.
  // https://github.com/maxogden/electron-packager/issues/150
  setTimeout(function() {
    if (!ncpCallbackFired) {
      cb(null);
    }
  }, 20000);
```

**Comment by davej** [comment]
Just want to mention that a number people have found ncp inconsistent and buggy and it isn't being actively maintained anymore: https://github.com/AvianFlu/ncp/issues/100

I also noticed that the fs-extra module dropped ncp because of bugs surrounding non-firing/double-firing of callbacks.

**Comment by maxogden** [comment]
we should switch to using https://www.npmjs.com/package/tar-fs: `tarfs.pack().pipe(tarfs.extract())`, its like `JSON.parse(JSON.stringify(obj))` but for filesystems :)

**Comment by malept** [comment]
We should open a new issue for that.

**Comment by jlord** [comment]
Looks like this was intended to be closed in #150.
